R/corpus.R

```diff
@@ -160,8 +165,8 @@ corpus.character <- function(x, docnames = NULL, docvars = NULL, ...) {
     names(x) <- docvar[["docname_"]]
     class(x) <- "corpus"
     attr(x, "unit") <- "documents"
-    attr(x, "meta") <- meta("character", ...)
     attr(x, "docvars") <- docvar
+    x <- meta_init(x, source = "character", user = meta)
```

Comment:
I think it is more flexible if  `meta_init()` only returns a list. Then 
```r
attr(x, "meta") <-  meta_init(source = "character", user = meta)
```
If the function is remaked to `make_meta()` then it becomes easier to understand what happening in functions:
```r
attr(x, "meta") <-  make_meta(...)
attr(x, "docvars") <-  make_docvars(...)
```

Reply:
Agreed, I will change this to a `meta()` that is exported and a `meta_system()/meta_system<-()` that work just like `meta()` but are not exported and that also has an initialize option. That way we can encapsulate all of the access to the attributes (or for a dfm, slots) inside `meta_system()`. Our goal should be only to access the attrs/slots there, not in any other function.